### PR TITLE
CBL-456 Set the level on Log.Console 

### PR DESCRIPTION
**Pull request by Sandychuang8**
CBL-456 Set the level on Log.Console and then set the domains according to what is included in the passed enum.

### CODE REVIEW

File: src/Couchbase.Lite.Shared/API/Database/Database.cs [hunk 1]
```diff
@@ -374,7 +374,9 @@ public static bool Exists([NotNull]string name, [CanBeNull]string directory)
         [ExcludeFromCodeCoverage]
 		public static void SetLogLevel(LogDomain domains, LogLevel level)
 		{
-		    WriteLog.To.Database.W(Tag, "Database.SetLogLevel is deprecated, use Database.Log.Console instead.");
+            Log.Console.Level = level;
+            Log.Console.Domains = domains;
+            WriteLog.To.Database.W(Tag, "Currently it will only affect the console logger and not the file logger. Please use Database.Log.Console instead if you need console logging.");
```
Comment (jayahariv) [inline src/Couchbase.Lite.Shared/API/Database/Database.cs]: indentation?

File: src/Couchbase.Lite.Shared/API/Database/Database.cs [hunk 2]
```diff
@@ -374,7 +374,9 @@ public static bool Exists([NotNull]string name, [CanBeNull]string directory)
         [ExcludeFromCodeCoverage]
 		public static void SetLogLevel(LogDomain domains, LogLevel level)
 		{
-		    WriteLog.To.Database.W(Tag, "Database.SetLogLevel is deprecated, use Database.Log.Console instead.");
+            Log.Console.Level = level;
```
Comment (jayahariv) [inline src/Couchbase.Lite.Shared/API/Database/Database.cs]: did you forget the indentation?

File: src/Couchbase.Lite.Tests.Shared/LogTest.cs [hunk 3]
```diff
@@ -291,6 +291,48 @@ public void TestConsoleLoggingLevels()
...
+            Database.SetLogLevel(LogDomain.Database, LogLevel.Verbose);
+            stringWriter = new StringWriter();
+            Console.SetOut(stringWriter);
+            WriteLog.To.Database.V("TEST", "TEST VERBOSE");
+            WriteLog.To.Database.I("TEST", "TEST INFO");
+            WriteLog.To.Database.W("TEST", "TEST WARNING");
+            WriteLog.To.Database.E("TEST", "TEST ERROR");
+            stringWriter.Flush();
+            stringWriter.ToString().Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries).Should()
+                .HaveCount(4, "because all levels should be logged");
```
Comment (jayahariv) [inline src/Couchbase.Lite.Tests.Shared/LogTest.cs]: i think you can include this logic inside the below foreach list, `new[] { LogLevel.Error, LogLevel.Warning, LogLevel.Info, LogLevel.Verbose}`

File: src/Couchbase.Lite.Tests.Shared/LogTest.cs [hunk 4]
```diff
@@ -291,6 +291,48 @@ public void TestConsoleLoggingLevels()
...
+                WriteLog.To.Database.W("TEST", "TEST WARNING");
+                WriteLog.To.Database.E("TEST", "TEST ERROR");
+                stringWriter.Flush();
+                stringWriter.ToString().Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries).Should()
+                    .HaveCount(currentCount, "because {0} levels should be logged for {1}", currentCount, level);
+                currentCount++;
+            }
+
+            Console.SetOut(new StreamWriter(Console.OpenStandardOutput()));
+            Database.SetLogLevel(LogDomain.Database, LogLevel.Warning);
```
Comment (jayahariv) [inline src/Couchbase.Lite.Tests.Shared/LogTest.cs]: are we going to enable all domain with warning level by default right? Seems like this only enabled database domain.

File: src/Couchbase.Lite.Shared/API/Database/Database.cs (hunk 2 above)
Comment (Sandychuang8) [reply src/Couchbase.Lite.Shared/API/Database/Database.cs]: The IDE automatically handles all the indentation.... They look fine in my codebase... but not sure what happens here...

Comment (borrrden) [reply src/Couchbase.Lite.Shared/API/Database/Database.cs]: Check tabs vs spaces....this is a very common problem.

File: src/Couchbase.Lite.Shared/API/Database/Database.cs [hunk 5]
```diff
@@ -373,10 +373,12 @@ public static bool Exists([NotNull]string name, [CanBeNull]string directory)
...
         [ExcludeFromCodeCoverage]
 		public static void SetLogLevel(LogDomain domains, LogLevel level)
-		{
-		    WriteLog.To.Database.W(Tag, "Database.SetLogLevel is deprecated, use Database.Log.Console instead.");
-		}
-        
+        {
+            Log.Console.Level = level;
+            Log.Console.Domains = domains;
+            WriteLog.To.Database.W(Tag, "Currently it will only affect the console logger and not the file logger.");
```
Comment (Sandychuang8) [inline src/Couchbase.Lite.Shared/API/Database/Database.cs]: Should we make this error message cross platform? Does this apply to all platform or just .Net?

Comment (borrrden) [reply src/Couchbase.Lite.Shared/API/Database/Database.cs]: This message is .NET specific.  The others never had this API affect the file logger (it didn't exist for them until the new log API was created)

File: src/Couchbase.Lite.Tests.Shared/LogTest.cs [hunk 6]
```diff
@@ -291,6 +291,47 @@ public void TestConsoleLoggingLevels()
...
+        {
+            WriteLog.To.Database.I("IGNORE", "IGNORE"); // Skip initial message
+            Database.SetLogLevel(LogDomain.All, LogLevel.None);
+            var stringWriter = new StringWriter();
+            Console.SetOut(stringWriter);
+            WriteLog.To.Database.E("TEST", "TEST ERROR");
+            stringWriter.Flush();
+            stringWriter.ToString().Should().BeEmpty("because logging is disabled");
+
+            Database.SetLogLevel(LogDomain.All, LogLevel.Verbose);
```
Comment (jayahariv) [inline src/Couchbase.Lite.Tests.Shared/LogTest.cs]: again, why are we keeping this out of the for loop below? is there a specific reason?

Comment (Sandychuang8) [reply src/Couchbase.Lite.Tests.Shared/LogTest.cs]: Different test scenario.

Comment (Sandychuang8) [reply src/Couchbase.Lite.Tests.Shared/LogTest.cs]: Will add log level verbose into the level array.

File: src/Couchbase.Lite.Tests.Shared/LogTest.cs [hunk 7]
```diff
@@ -291,6 +291,36 @@ public void TestConsoleLoggingLevels()
             Database.Log.Console.Level = LogLevel.Warning;
         }
 
+        [Fact] void TestSetLogLevel()
```
Comment (borrrden) [inline src/Couchbase.Lite.Tests.Shared/LogTest.cs]: Put newline after `[Fact]`

**Pull request merged**